File: app/src/main/java/com/fos/sample/ui/login/fragment/LoginFragment.java
package com.fos.sample.ui.login.fragment;

import android.os.Bundle;

import com.fos.fosmvp.common.base.BaseResponse;
import com.fos.fosmvp.common.utils.ToastUtils;
import com.fos.sample.R;
import com.fos.sample.base.BaseOptionFragment;
import com.fos.sample.entity.login.UserEntity;
import com.fos.sample.ui.login.contract.LoginContract;
import com.fos.sample.ui.login.model.LoginModel;
import com.fos.sample.ui.login.presenter.LoginPresenter;

/**
 * Fragment使用示例
 *
 */
public class LoginFragment extends BaseOptionFragment<LoginPresenter, LoginModel> implements LoginContract.View{

    public String tel = "";
    public String password = "";

    @Override
    protected int getLayoutResource() {
        return R.layout.activity_login;
    }

    @Override
    public void initPresenter() {
        setPresenter();
    }

    @Override
    protected void initView(Bundle savedInstanceState) {

    }



    @Override
    public void returnLoginSucceed(UserEntity userEntity) {
        ToastUtils.showShort("");
    }


    @Override
    public void returnLoginFail(BaseResponse baseResponse, boolean isVisitError) {
        try {
            if (!isVisitError){
                ToastUtils.showShort(baseResponse.getMsg());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/fos/sample/login/fragment/LoginFragment.kt
package com.fos.sample.ui.login.fragment


import android.os.Bundle
import android.view.View
import com.fos.fosmvp.common.base.BaseFragment
import com.fos.fosmvp.common.base.BaseResponse
import com.fos.fosmvp.common.utils.LogUtils
import com.fos.fosmvp.common.utils.ToastUtils
import com.fos.sample.R
import com.fos.sample.entity.login.UserEntity
import com.fos.sample.ui.login.contract.LoginContract
import com.fos.sample.ui.login.model.LoginModel
import com.fos.sample.ui.login.presenter.LoginPresenter

/**
 * Fragment使用示例
 *
 */
class LoginFragment : BaseFragment<LoginPresenter, LoginModel>(), LoginContract.View,View.OnClickListener {
    var tel = ""
    var password = ""

    override val layoutResource: Int
        get() = R.layout.activity_login

    override fun initPresenter() {
        mPresenter!!.setViewModel(this, mModel!!)
    }

    override fun initView(savedInstanceState: Bundle?) {

    }

    override fun onClick(view: View) {

    }

    fun login() {

        LogUtils.e("tel= "+tel)
        ToastUtils.showShort("登录")
        mPresenter!!.getLoginRequest(tel, password)
    }

    override fun returnLoginSucceed(userEntity: UserEntity) {
        ToastUtils.showShort("登录成功")
    }


    override fun returnLoginFail(baseResponse: BaseResponse<*>?, isVisitError: Boolean) {
        try {
            if (!isVisitError&&baseResponse!=null) {
                ToastUtils.showShort(baseResponse.msg!!)
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }
}
